packages/@n8n/nodes-langchain/nodes/chains/ChainLLM/ChainLlm.node.ts

```diff
@@ -106,44 +115,53 @@ export class ChainLlm implements INodeType {
 					[],
 				) as MessageTemplate[];
 
-				// Execute the chain
-				const responses = await executeChain({
+				return (await executeChain({
 					context: this,
 					itemIndex,
 					query: prompt,
 					llm,
 					outputParser,
 					messages,
-				});
+				})) as object[];
+			});
 
-				// If the node version is 1.6(and LLM is using `response_format: json_object`) or higher or an output parser is configured,
-				//  we unwrap the response and return the object directly as JSON
-				const shouldUnwrapObjects = this.getNode().typeVersion >= 1.6 || !!outputParser;
-				// Process each response and add to return data
-				responses.forEach((response) => {
-					returnData.push({
-						json: formatResponse(response, shouldUnwrapObjects),
-					});
-				});
-			} catch (error) {
-				// Handle OpenAI specific rate limit errors
-				if (error instanceof NodeApiError && isOpenAiError(error.cause)) {
-					const openAiErrorCode: string | undefined = (error.cause as any).error?.code;
-					if (openAiErrorCode) {
-						const customMessage = getCustomOpenAiErrorMessage(openAiErrorCode);
-						if (customMessage) {
-							error.message = customMessage;
+			const batchResults = await Promise.allSettled(batchPromises);
+
+			batchResults.forEach((promiseResult, batchItemIndex) => {
+				const itemIndex = i + batchItemIndex;
+				if (promiseResult.status === 'rejected') {
+					const error = promiseResult.reason as Error;
+					// Handle OpenAI specific rate limit errors
+					if (error instanceof NodeApiError && isOpenAiError(error.cause)) {
+						const openAiErrorCode: string | undefined = (error.cause as any).error?.code;
+						if (openAiErrorCode) {
+							const customMessage = getCustomOpenAiErrorMessage(openAiErrorCode);
+							if (customMessage) {
+								error.message = customMessage;
+							}
 						}
 					}
-				}
 
-				// Continue on failure if configured
-				if (this.continueOnFail()) {
-					returnData.push({ json: { error: error.message }, pairedItem: { item: itemIndex } });
-					continue;
+					if (this.continueOnFail()) {
+						returnData.push({
+							json: { error: error.message },
+							pairedItem: { item: itemIndex },
+						});
+						return;
+					}
+					throw new NodeOperationError(this.getNode(), error);
 				}
 
-				throw error;
+				const responses = promiseResult.value;
+				responses.forEach((response: object) => {
+					returnData.push({
+						json: formatResponse(response, this.getNode().typeVersion >= 1.6 || !!outputParser),
```

Comment:
Maybe leave the comment that was on line 116 about how the node version and different unwrapping.